### PR TITLE
[internal][tracing] Validate `trace.provider` in pf config

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

This PR targets to validate `trace.provider` value:

1. valid ARM resource id for Azure ML workspace
2. the workspace exists
3. the workspace corresponding Cosmos DB is initialized

image

While Azure CLI is not login, clear error message will be raised from `MLClient` init

image

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by crazygao — APPROVED** [review]

File: src/promptflow/promptflow/azure/_utils/_tracing.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        return DefaultAzureCredential()
+
+
+def _create_trace_provider_value_user_error(
+    message: str, no_personal_data_message: typing.Optional[str] = None
+) -> UserErrorException:
+    return UserErrorException(
+        message=message,
+        target=ErrorTarget.CONTROL_PLANE_SDK,
+        no_personal_data_message=no_personal_data_message,
```
Comment (wangchao1230) [inline src/promptflow/promptflow/azure/_utils/_tracing.py:34]: what's this? no_personal_data_message

Comment (zhengfeiwang) [reply src/promptflow/promptflow/azure/_utils/_tracing.py:34]: i undertstand this field will be collected by our telemetry as there is no personal data here?

\+ @D-W- and @Stephen1993 for precise explanation, not sure if i'm using it incorrectly.

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/promptflow/azure/_utils/_tracing.py (hunk 1 above)
Comment (zhengfeiwang) [reply src/promptflow/promptflow/azure/_utils/_tracing.py]: seems i'm using memory from v2... just removed this.

**Review by brynn-code — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

File: src/promptflow/promptflow/azure/_utils/_tracing.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        return credential
+    except Exception:
+        return DefaultAzureCredential()
+
+
+def _create_trace_provider_value_user_error(message: str) -> UserErrorException:
+    return UserErrorException(message=message, target=ErrorTarget.CONTROL_PLANE_SDK)
+
+
+def validate_trace_provider(value: str) -> None:
```
Comment (0mza987) [inline src/promptflow/promptflow/azure/_utils/_tracing.py:32]: Consider adding a test?

**Pull request merged**